### ISSUE TITLE
jax.random.categorical doesn't respect shape parameter with broadcasted logits

### ISSUE THREAD

**Issue by dlwh** [google/jax#13124]
### Description

The docs for `jax.random.categorical` say that the output of `jax.random.categorical` is `shape` when `shape` is provided. However, when provided `logits` which have to be broadcasted with `shape`, it just ignores the shape parameter after checking that it's broadcast compatible. I believe this can be fixed by broadcasting logits appropriately.

```
import jax
out = jax.random.categorical(jax.random.PRNGKey(0), logits=jax.numpy.ones( (4, 1, 10)), axis=-1, shape=(4, 8))

# out.shape is (4, 1), should be (4, 8)

assert out.shape == (4, 8), f"out.shape is {out.shape}, should be (4, 8)"
```

### What jax/jaxlib version are you using?

0.3.23

### Which accelerator(s) are you using?

CPU

**Comment by mattjj** [comment]
Thanks for raising this, and for the clear diagnosis! #13127 should fix.

**Comment by dlwh** [comment]
thanks for fixing so quickly!
